client/src/features/ProjectPageV2/ProjectPageContent/Settings/ProjectSettingsMembers.tsx

```diff
@@ -0,0 +1,348 @@
+    <div className={cx("bg-white", "rounded-3", "mt-3", "p-3", "p-md-4")}>
+      {children}
+    </div>
+  );
+}
+
+type MemberActionMenuProps = Omit<
+  ProjectPageSettingsMembersTableRowProps,
+  "member" | "members"
+> & { disabled?: boolean };
+
+function MemberActionMenu({
+  disabled,
+  index,
+  onRemove,
+  onEdit,
+}: MemberActionMenuProps) {
+  const defaultAction = (
+    <Button
+      color="rk-green"
+      disabled={disabled}
+      data-cy={`project-member-edit-${index}`}
+      onClick={onEdit}
+    >
+      <PencilSquare className={cx("rk-icon", "rk-icon-sm", "me-2")} /> Edit
+    </Button>
+  );
+  return (
+    <ButtonWithMenu
+      className="py-1"
+      color="rk-green"
+      default={defaultAction}
+      disabled={disabled}
+      isPrincipal
+    >
+      <DropdownItem onClick={onRemove}>
+        <Trash className={cx("rk-icon", "rk-icon-sm", "me-2")} /> Remove
+      </DropdownItem>
+    </ButtonWithMenu>
+  );
+}
+
+interface ProjectPageSettingsMembersTableRowProps {
+  index: number;
+  member: ProjectMemberResponse;
+  members: ProjectMemberResponse[];
+  onRemove: () => void;
+  onEdit: () => void;
+}
+
+function ProjectPageSettingsMembersTableRow({
+  index,
+  member,
+  members,
+  onRemove,
+  onEdit,
+}: ProjectPageSettingsMembersTableRowProps) {
+  return (
+    <Row className={cx("px-0", "py-4", "py-xl-3", "m-0")}>
+      <Col
+        xl={6}
+        xs={12}
+        className={cx("d-flex", "align-items-center", "px-3")}
+      >
+        {member.email ?? member.id}
+      </Col>
+      <Col
+        xl={3}
+        sm={6}
+        xs={12}
+        className={cx("d-flex", "align-items-center", "px-2")}
+      >
+        {member.role}
+      </Col>
+      <Col
+        xl={3}
+        sm={6}
+        xs={12}
+        className={cx("d-flex", "align-items-center", "px-2")}
+        data-cy={`project-member-actions-${index}`}
+      >
+        <MembershipGuard
+          disabled={
+            <MemberActionMenu
+              disabled={true}
+              index={index}
+              onRemove={onRemove}
+              onEdit={onEdit}
+            />
+          }
+          enabled={
+            <MemberActionMenu
+              index={index}
+              onRemove={onRemove}
+              onEdit={onEdit}
+            />
+          }
+          members={members}
+        />
+      </Col>
+    </Row>
+  );
+}
+
+interface ProjectPageSettingsMembersTableProps {
+  members: ProjectMemberResponse[];
+  projectId: string;
+}
+
+function ProjectPageSettingsMembersTable({
+  members,
+  projectId,
+}: ProjectPageSettingsMembersTableProps) {
+  const [isEditMemberModalOpen, setIsEditMemberModalOpen] = useState(false);
+  const [isRemoveMemberModalOpen, setIsRemoveMemberModalOpen] = useState(false);
+  const [memberToEdit, setMemberToEdit] = useState<ProjectMemberResponse>();
+  const sortedMembers = [...members].sort((a, b) => {
+    if (a.role !== b.role) {
+      return roleCompare(a.role, b.role, false);
```

Comment:
This is unnecessarily hard to understand.

```suggestion
      //? Descending order of roles
      return toNumericRole(b.role) - toNumericRole(a.role);
```

See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#comparefn
> To memorize this, remember that (a, b) => a - b sorts numbers in ascending order.

Reply:
Addressed in cbda6a744409a0d51b978900e319cf0151162ed1